### PR TITLE
Add currently processing status

### LINKED ISSUE #214: Add in process status
Add a currently processing status to improve performance and eliminate running a service multiple times over the same set of files. This is similar to recent changes to the `BaseRoutine` in Zorya and Pictor.

### PR DESCRIPTION
**Pull request by bonniegee**
Fixes #214

### CODE REVIEW

File: sip_assembly/routines.py [hunk 1]
```diff
@@ -56,16 +56,23 @@ class BaseRoutine(object):
...
-            except Exception as e:
-                raise Exception(str(e), sip.bag_identifier)
+                try:
+                    message = self.process_sip(sip)
+                    sip.process_status = self.end_status
+                    sip.save()
+                except Exception as e:
+                    sip.process_status = self.start_status
+                    sip.save()
+                    raise Exception(str(e), sip.bag_identifier)
```
Comment (helrond) [inline sip_assembly/routines.py]: I think how you did this in Pictor is better:

```suggestion
                try:
                    message = self.process_sip(sip)
                except Exception as e:
                    sip.process_status = self.start_status
                    sip.save()
                    raise Exception(str(e), sip.bag_identifier)
                 sip.process_status = self.end_status
                 sip.save()
```

**Review by helrond — CHANGES_REQUESTED** [review]
One small tweak to more specifically target try/except. Also I think you need a migration?

**Comment by bonniegee** [comment]
> One small tweak to more specifically target try/except. Also I think you need a migration?

Perhaps I am doing something wrong but literally nothing happens when I run `makemigrations` 🤔

**Comment by helrond** [comment]
Hmm okay I would have thought that altering choices for a field would require a migration, but perhaps I am wrong!

**Review by helrond — APPROVED** [review]

**Comment by bonniegee** [comment]
@helrond should I merge or wait for #215?

**Comment by helrond** [comment]
I think you can merge - this is pretty low-risk.

**Pull request merged**